histogram/src/snapshot.rs

```diff
@@ -32,6 +32,20 @@ impl Snapshot {
         self.histogram.percentile(percentile)
     }
 
+    /// Returns a new downsampled histogram with a reduced grouping power.
+    ///
+    /// The new histogram is smaller but with greater relative error. The
+    /// reduction factor should be smaller than the histogram's existing
+    /// grouping power.
+    pub fn downsample(&self, factor: u8) -> Result<Snapshot, BuildError> {
```

Comment:
I think this should be `Error` and not build error? I think the build errors should only occur when constructing a histogram and otherwise we return the Error type at runtime.

Reply:
Fair enough, changed, though its murky whether this is an operation or a construction. Are we really benefiting from having 2 different types of errors?

Reply:
I agree, we might want to merge the two error types. We should leave that to a separate PR.